src/main/telemetry/srxl.c

```diff
@@ -462,6 +469,16 @@ static void processSrxl(timeUs_t currentTimeUs)
     } else {
         srxlFnPtr = srxlScheduleFuncs[srxlScheduleIndex + srxlScheduleUserIndex];
         srxlScheduleUserIndex = (srxlScheduleUserIndex + 1) % SRXL_SCHEDULE_USER_COUNT;
+
+#if defined (USE_SPEKTRUM_CMS_TELEMETRY) && defined (USE_CMS)
+        // Boost CMS performance by sending nothing else but CMS Text frames when in a CMS menu.
+        // Sideeffect, all other reports are still not sent if user leaves CMS without a proper EXIT.
+        if (cmsInMenu &&
+            (pCurrentDisplay == &srxlDisplayPort)) {
```

Comment:
I think this check is needed anyway, we should not be processing data if it's not for the SRXL display port?

Reply:
Exactly the reason I added this check. When having CMS on OSD, no reason to sub-optimize telemetry. Only when CMS is running on the actual srxl display.